src/index.js

```diff
@@ -1,91 +1,109 @@
 'use strict'
 
-const parallelLimit = require('async/parallelLimit')
-const pull = require('pull-stream')
-
-// BlockService is a hybrid block datastore. It stores data in a local
-// datastore and may retrieve data from a remote Exchange.
-// It uses an internal `datastore.Datastore` instance to store values.
-module.exports = class BlockService {
+/**
+ * BlockService is a hybrid block datastore. It stores data in a local
+ * datastore and may retrieve data from a remote Exchange.
+ * It uses an internal `datastore.Datastore` instance to store values.
+ */
+class BlockService {
+  /**
+   * Create a new BlockService
+   *
+   * @param {IPFSRepo} ipfsRepo
+   * @returns {BlockService}
+   */
   constructor (ipfsRepo) {
     this._repo = ipfsRepo
     this._bitswap = null
   }
 
+  /**
+   * Add a bitswap instance that communicates with the
+   * network to retreive blocks that are not in the local store.
+   *
+   * If the node is online all requests for blocks first
+   * check locally and afterwards ask the network for the blocks.
+   *
+   * @param {Bitswap} bitswap
+   * @returns {void}
+   */
   goOnline (bitswap) {
     this._bitswap = bitswap
   }
```

Comment:
This method name is a bit misleading. By passing bitswap, we are not necessarily going online (i.e bitswap is offline).

What about `addExchange`, so that in the future, we even just have to add multiple exchanges and it all works.